amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/EmptyLeaseTableSynchronizer.java

```diff
@@ -0,0 +1,78 @@
+package software.amazon.kinesis.leases;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.services.kinesis.model.Shard;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.leases.exceptions.DependencyException;
+import software.amazon.kinesis.leases.exceptions.InvalidStateException;
+import software.amazon.kinesis.leases.exceptions.ProvisionedThroughputException;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static software.amazon.kinesis.leases.HierarchicalShardSyncer.constructShardIdToShardMap;
+import static software.amazon.kinesis.leases.HierarchicalShardSyncer.convertToCheckpoint;
+import static software.amazon.kinesis.leases.HierarchicalShardSyncer.newKCLLease;
+
+@Slf4j
+@AllArgsConstructor
+public class EmptyLeaseTableSynchronizer implements LeaseSynchronizer {
+
+    @Override
+    public List<Lease> determineNewLeasesToCreate(List<Shard> shards, List<Lease> currentLeases,
+                                                  InitialPositionInStreamExtended initialPosition,
+                                                  Set<String> inconsistentShardIds,
+                                                  HierarchicalShardSyncer.MultiStreamArgs multiStreamArgs) {
+
+        final Map<String, Shard> shardIdToShardMapOfAllKinesisShards = constructShardIdToShardMap(shards);
+
+        currentLeases.stream().peek(lease -> log.debug("Existing lease: {}", lease))
+                .map(lease -> shardIdFromLeaseDeducer.apply(lease, multiStreamArgs))
+                .collect(Collectors.toSet());
+
+        final List<Lease> newLeasesToCreate = getLeasesToCreateForOpenAndClosedShards(initialPosition, shards);
+
+        final Comparator<Lease> startingSequenceNumberComparator =
+                new HierarchicalShardSyncer.StartingSequenceNumberAndShardIdBasedComparator(shardIdToShardMapOfAllKinesisShards, multiStreamArgs);
+        newLeasesToCreate.sort(startingSequenceNumberComparator);
+        return newLeasesToCreate;
+    }
+
+    @Override
+    public void cleanupGarbageLeases(List<Shard> shards, LeaseRefresher leaseRefresher, List<Lease> trackedLeases, HierarchicalShardSyncer.MultiStreamArgs multiStreamArgs) throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        // Nothing to do here.
+    }
+
+    @Override
+    public void cleanupLeasesOfFinishedShards(LeaseRefresher leaseRefresher, List<Lease> currentLeases, List<Lease> trackedLeases, HierarchicalShardSyncer.MultiStreamArgs multiStreamArgs) throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        // Nothing to do here.
+    }
```

Comment:
I see, didn't realize we were moving the GC to periodic shard sync. Will see if we can restrict this away from non-period shard sync cases

Reply:
Ended up removing these from the interface, I think just using this interface to strictly create leases will be less invasive; we can just pass a parameter to garbage collect or not so that periodic shard sync can opt into garbage collecting.